src/yoti_common/anchor.processor.js

```diff
@@ -0,0 +1,301 @@
+'use strict';
+
+const forge = require('node-forge');
+const protoRoot = require('../proto-root');
+
+const YotiAnchor = function main(anchorObj) {
+  this.value = anchorObj.value;
+  this.subType = anchorObj.subType;
+  this.signedTimeStamp = anchorObj.signedTimeStamp;
+  this.originServerCerts = anchorObj.originServerCerts;
+};
+
+YotiAnchor.prototype = {
+  getValue() { return this.value; },
+  getSubType() { return this.subType; },
+  getSignedTimeStamp() { return this.signedTimeStamp; },
+  getOriginServerCerts() { return this.originServerCerts; },
+};
+
+const YotiSignedTimeStamp = function main (timeStampObj) {
+  this.version = timeStampObj.version;
+  this.timestamp = timeStampObj.timestamp;
+};
+
+YotiSignedTimeStamp.prototype = {
+  getVersion() { return this.version; },
+  getTimestamp() { return this.timestamp; }
+};
+
+module.exports.AnchorProcessor = class AnchorProcessor {
+  /**
+   * Extract matching Attribute Anchors from list.
+   *
+   * @param anchors
+   *
+   * @returns {Array}
+   */
+  static process(anchors) {
+    // This will contain a list of anchor values as objects
+    let anchorsData = this.getResultFormat();
+    for (let i = 0; i < anchors.length; i += 1) {
+      const yotiAnchorsList = this.processSingleAnchor(anchors[i]);
+      anchorsData = this.mergeAnchorsList(anchorsData, yotiAnchorsList);
+    }
+    return anchorsData;
+  }
+
+  /**
+   * Return YotiAnchors list from protobuf anchor.
+   *
+   * @param anchorObj
+   *
+   * @returns {Array}
+   */
+  static processSingleAnchor(anchorObj) {
+    let anchorsList = this.getResultFormat();
+
+    if (!anchorObj instanceof Object) {
+      return anchorsList;
+    }
+
+    const certificatesList = anchorObj.originServerCerts;
+    const yotiSignedTimeStamp = this.processSignedTimeStamp(anchorObj.getSignedTimeStamp());
+    const serverX509Certs = AnchorProcessor.convertCertsListToX509(anchorObj.originServerCerts);
+    const subType = anchorObj.getSubType();
+
+    for (let j = 0; j < certificatesList.length; j += 1) {
+      const certAnchors = this.getOidsAnchorsByCertificate(
+        certificatesList[j],
+        subType,
+        yotiSignedTimeStamp,
+        serverX509Certs
+      );
+      anchorsList = this.mergeAnchorsList(anchorsList, certAnchors);
+    }
+    return anchorsList;
+  }
+
+  /**
+   * Get Oids anchors from a single certificate.
+   *
+   * @param certArrayBuffer
+   * @param subType
+   * @param yotiSignedTimeStamp
+   * @param X509Certs
+   *
+   * @returns {Array}
+   */
+  static getOidsAnchorsByCertificate(certArrayBuffer, subType, yotiSignedTimeStamp, X509Certs) {
```

Comment:
I think this can be just `getAnchorsByCertificate` as the name should say what the function does, instead of how.
Much more readable now btw.